Assignment 3.Rmd

```diff
@@ -136,18 +153,79 @@ What patterns do you see in the plot?
 It would be useful to determine how many people are in each cluster. We can do this easily with dplyr.
 
 ```{r}
-K7 <- 
+K5$cluster <- as.factor(K5$cluster)
+K7 <- K4 %>% 
+  group_by(cluster) %>% 
+  summarize(count = n())
 ```
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
 
+```{r}
+fit2 <- kmeans(K3, 3)
+K4_2 <- data.frame(K3, fit2$cluster)
+names(K4_2) <- c("1", "2", "3", "4", "5", "cluster")
+K5_2 <- K4_2 %>% 
+  gather(week, mtv_value, 1:5)
+K6_2 <- K5_2 %>%
+  group_by(cluster, week) %>% 
+  summarise(avg_mtv = mean(mtv_value)) %>% 
+  arrange(avg_mtv)
+
+K6_2$week <- as.factor(K6_2$week)
+
+K6_2$cluster <- as.factor(K6_2$cluster)
+
+K6_2$week <- as.numeric(K6_2$week)
+f2 <- ggplot(K6_2, aes(week, avg_mtv, color = cluster)) + geom_line()
+f
+f2
+fviz_nbclust(K3, kmeans, method = "gap_stat")
+
+```
+
+
 Once you have done this, save both of your plots to the Assignment 5 file. Create a Zotero item to index your new computer program (Assignment 5.rmd) in Zotero. Then commit your assignment, push it to your Github account and then Pull request your version to the original assignment version so I can see it.
 
 ##Part II
 
-Now, try to do the same for the data collected in class. Create two sets of clusters, the answers to the questions and regions where people grew up. 
+Now, try to do the same for the data collected in class. Create two sets of clusters, the answers to the questions and regions where people grew up.
+```{r}
+library(dplyr)
+library(tidyr)
+df <- read.csv("HUDK405018-cluster.csv", na.strings=c("", "NA"))
+df1 <- df[-c(1,2), ]
+place <- df1[ , 18:19]
+answer <- df1[ , 20:26]
+answer2 <- na.omit(answer)
+answer3 <- sapply( answer2, as.numeric )
+
+library(ggmap)
+library(cluster)
+df_ll <- geocode(as.character(place$Q1_1), output = "latlon", source = "dsk")
+
+
+# how to determin k in kmeans?
+fviz_nbclust(df_ll, kmeans, method = "gap_stat")
+fviz_nbclust(df_ll, kmeans, method = "wss")
+fviz_nbclust(df_ll, kmeans, method = "silhouette")
+
+fviz_nbclust(answer3, kmeans, method = "gap_stat")
+fviz_nbclust(answer3, kmeans, method = "wss")
+fviz_nbclust(answer3, kmeans, method = "silhouette")
+
```

Comment:
Glad to see you thinking about the appropriate number of clusters. Maybe provide some interpretation of the different methods.